src/Particles/babylon.solidParticleSystem.ts

```diff
@@ -337,50 +300,34 @@ module BABYLON {
         private _unrotateFixedNormals() {
             var index = 0;
             var idx = 0;
+            const tmpNormal = Tmp.Vector3[0];
+            const rotMatrix = Tmp.Matrix[0];
+            const invertedRotMatrix = Tmp.Matrix[1];
             for (var p = 0; p < this.particles.length; p++) {
-                this._particle = this.particles[p];
-                this._shape = this._particle._model._shape;
-                if (this._particle.rotationQuaternion) {
-                    this._quaternion.copyFrom(this._particle.rotationQuaternion);
-                }
-                else {
-                    this._yaw = this._particle.rotation.y;
-                    this._pitch = this._particle.rotation.x;
-                    this._roll = this._particle.rotation.z;
-                    this._quaternionRotationYPR();
-                }
-                this._quaternionToRotationMatrix();
-                this._rotMatrix.invertToRef(this._invertMatrix);
+                const particle = this.particles[p];
+                const shape = particle._model._shape;
+                particle.getRotationMatrix(rotMatrix);
+                rotMatrix.invertToRef(invertedRotMatrix);

```

Comment:
@jbousquie Here it might be faster to retrieve the matrix of the conjugate quaternion than inverting a matrix (the result should be the same I think)

Reply:
well, I don't know anything about conjugate quaternions. I just read how they are computed, what seems to be fast indeed. Then, I have no idea what this conjugate means and how to use it in order to "un-rotate" a given vector. My 3D maths aren't enough here but I'm still willing to learn. Do you have a link about something explaining how to use the conjugate quaternion ?